### PR TITLE
feat(oidc): allow to declare domains used by OIDC server and domains to which access token is appended separately

### PR DESCRIPTION
**Pull request by krzempekk**
This implements #1052.

## Before this PR

There were only two ways to define `TrustedDomains` config:
```
const trustedDomains = {
    default: ["https://demo.duendesoftware.com"],
};
```
```
const trustedDomains = {
    default: { domains : ["https://demo.duendesoftware.com"], showAccessToken: true },
};
```
It was not possible to define domains used by OIDC separately from domains to which access token should be injected.

## After this PR

There is new way to define `TrustedDomains` config:
```
const trustedDomains = {
    default: { 
         oidcDomains : ["https://demo.duendesoftware.com"], 
         accessTokenDomains : ["https://myapi"], 
         showAccessToken: true 
    },
};
```
This way allows to define domains used by OIDC separately from domains to which access token should be injected.
If both `domains` and `oidcDomains` are defined, `oidcDomains` takes precedence. 
If both `domains` and `accessTokenDomains` are defined, `accessTokenDomains` takes precedence.

### CODE REVIEW

File: packages/react/service_worker/OidcServiceWorker.ts [hunk 1]
```diff
@@ -321,8 +322,12 @@ const handleMessage = (event: ExtendableMessageEvent) => {
     case 'init': {
       const oidcServerConfiguration = data.data.oidcServerConfiguration;
       let trustedDomain = trustedDomains[configurationName];
-      
-      const domains = Array.isArray(trustedDomain) ? trustedDomain : trustedDomain.domains;
+      let domains: Domain[];
```
Comment (darkowic) [inline packages/react/service_worker/OidcServiceWorker.ts:325]: Consider extracting the logic to a function

**Comment by guillaume-chervet** [comment]
Hi @krzempekk , i have clone your fork, i finish to look at it tomorrow.
Thank you for your help !

File: packages/react/service_worker/OidcServiceWorker.ts (hunk 1 above)
Comment (krzempekk) [reply packages/react/service_worker/OidcServiceWorker.ts]: Extracted to one function handling both cases

**Comment by krzempekk** [comment]
Hi @guillaume-chervet !
No worries, thank you very much for an update

File: readme.md [hunk 2]
```diff
@@ -173,6 +173,12 @@ const trustedDomains = {
 // scenarios which require it. For example, to send it via websocket connection.
 trustedDomains.config_show_access_token = { domains : ["https://demo.duendesoftware.com"], showAccessToken: true };
 
+// This example defines domains used by OIDC server separately from domains to which access tokens will be injected.
+trustedDomains.config_separate_oidc_access_token_domains = {
```
Comment (guillaume-chervet) [inline readme.md:177]: hi @krzempekk , it woulob be nice to add this sample in multiAuthDemo page.

**Review by guillaume-chervet — APPROVED** [review]

**Comment by guillaume-chervet** [comment]
thank you very much @krzempekk , I really like this separation :)

tell me went you are ready to merge !

File: readme.md (hunk 2 above)
Comment (krzempekk) [reply readme.md:177]: Done

**Comment by krzempekk** [comment]
Hi @guillaume-chervet ! Thank you very much for review. I added sample to multiAuthDemo page. I think PR is ready to merge now :)

**Pull request merged**